### PR TITLE
:hourglass: run on schedule, deploy to rancher

### PR DESCRIPTION
**Pull request by markysoft**
## Checklist

- [ ] An [ADR](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions)
- [ ] New and/or updated tests
- [x] Changes log in `CHANGELOG`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,3 +10,6 @@ before_script: greenkeeper-lockfile-update
 script: bash ./scripts/test-ci
 
 after_script: docker-compose -f docker-compose-tests.yml down -v
+
+after_success:
+  - bash ./scripts/deploy.sh
```
Comment (st3v3nhunt) [inline .travis.yml]: Can this be all on one line for consistency with the rest of the file please?

File: README.md [hunk 2]
```diff
@@ -64,9 +73,15 @@ The files uploaded to Azure Blob Storage are:
 where `YYYYMMDD` is the current year, month and date, and `VERSION` is the current major version of the ETL as defined in the `package.json`.
 
 ### Development Notes
+The ETL may be configured to collect data from the 'Sexual Health Information and Support Services', or the 'Chlamydia Screening for Under 25s'
+end point in Syndication. The details above describe the operation when configuted to retrieve 'Sexual Health Information and Support Services' data.
```
Comment (st3v3nhunt) [inline README.md]: Typo `configuted => configured`

File: README.md [hunk 3]
```diff
@@ -107,13 +122,26 @@ environment.
...
+
+The `docker-compose.yml` used for development and deployment via Rancher have a similar structure.
+A stack is run with two `sexual-health-information-etl` images with different configurations.
+
+The convention for environment variables used in the Rancher configuration is to add a `SHIS_` or `CSU25_` prefix to the
+`ETL_SCHEDULE`, `SYNDICATION_SERVICE_END_POINT` and `OUTPUT_FILE` environment variables.
+These are then mapped to the appropriate suffix-less variable in the container, i.e. for the
+'Sexual Health Information and Support Services' container `SHIS_ETL_SCHEDULE` is mapped to `ETL_SCHEDULE`, and so on.
+
+The default
```
Comment (st3v3nhunt) [inline README.md]: Unfinished sentence needs sorting out please.

File: docker-compose.yml [hunk 4]
```diff
@@ -15,12 +15,32 @@ services:
...
       SYNDICATION_SERVICE_END_POINT: http://v1.syndication.nhschoices.nhs.uk/services/types/sexualhealthinformationandsupport
       OUTPUT_FILE: ${OUTPUT_FILE-dev-shis-data}
+      ETL_SCHEDULE: ${ETL_SCHEDULE}
 
-  etl:
+  shis-etl:
     extends:
       service: base-etl
+    environment:
+      ETL_NAME: shis-data-etl
```
Comment (st3v3nhunt) [inline docker-compose.yml]: Could the env vars be alphabetised please (for all 3 services ideally)?

File: rancher-config/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+version: '2'
+services:
+  shis-etl:
+    image: nhsuk/sexual-health-information-etl:${DOCKER_IMAGE_TAG}
+    environment:
+      ETL_NAME: 'shis-data-etl'
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: Could the env vars be alphabetised please (and the same for `csu25`)?

File: schedule.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+const schedule = require('node-schedule');
+
+const log = require('./app/lib/logger');
```
Comment (st3v3nhunt) [inline schedule.js]: Can the requires be alphabetised please?

File: docker-compose.yml [hunk 7]
```diff
@@ -15,12 +15,32 @@ services:
       SYNDICATION_API_KEY: ${SYNDICATION_API_KEY}
       SYNDICATION_SERVICE_END_POINT: http://v1.syndication.nhschoices.nhs.uk/services/types/sexualhealthinformationandsupport
```
Comment (st3v3nhunt) [inline docker-compose.yml]: Should this have a default now or could it just be `${SYNDICATION_SERVICE_END_POINT}`?
Same goes for `ETL_NAME` above.

File: docker-compose.yml [hunk 8]
```diff
@@ -15,12 +15,32 @@ services:
...
       OUTPUT_FILE: ${OUTPUT_FILE-dev-shis-data}
```
Comment (st3v3nhunt) [inline docker-compose.yml]: Should this have a default or would `${OUTPUT_FILE}` be ok now?

File: rancher-config/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+      INITIAL_LAST_RUN_DATE: '${INITIAL_LAST_RUN_DATE}'
+      AZURE_STORAGE_CONNECTION_STRING: '${AZURE_STORAGE_CONNECTION_STRING}'
+      CONTAINER_NAME: ${CONTAINER_NAME}
+      SYNDICATION_API_KEY: ${SYNDICATION_API_KEY}
+      SYNDICATION_SERVICE_END_POINT: ${SHIS_SYNDICATION_SERVICE_END_POINT}
+      OUTPUT_FILE: ${SHIS_OUTPUT_FILE}
+      DISABLE_SCHEDULER: ${DISABLE_SCHEDULER}
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: Would one ETL want to be disabled without the other?

File: rancher-config/docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,55 @@
+version: '2'
+services:
+  shis-etl:
+    image: nhsuk/sexual-health-information-etl:${DOCKER_IMAGE_TAG}
+    environment:
+      ETL_NAME: 'shis-data-etl'
+      INITIAL_LAST_RUN_DATE: '${INITIAL_LAST_RUN_DATE}'
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: Would one ETL want to have a different last run date than the other?

Comment (markysoft) [reply rancher-config/docker-compose.yml]: Potentially, I was under the impression that everything in these buckets were uploaded from the same source sheet, but I'm not certain so will expose them as individual variables.

File: docker-compose.yml (hunk 7 above)
Comment (markysoft) [reply docker-compose.yml]: I've removed the defaults from here and explicitly set the required one in the docker-compose-test.yml.

**Pull request merged**